src/query/catalog/src/table_context.rs

```diff
@@ -203,12 +203,13 @@ pub trait TableContext: Send + Sync {
     fn get_current_session_id(&self) -> String {
         unimplemented!()
     }
-    async fn get_all_effective_roles(&self) -> Result<Vec<RoleInfo>>;
+    async fn get_all_effective_roles(&self, create_object: bool) -> Result<Vec<RoleInfo>>;
```

Comment:
maybe we do not need add extra parameter on this method? we can consider CREATE XXX is a special case, and only make special checks with current role inside these operations, and do not need others to care about this special case.

Reply:
we can consider move this logic out of `get_all_effective_roles`, and calls `get_current_role` on this special case.